site/src/App/Navigation/Navigation.tsx

```diff
@@ -1,96 +1,162 @@
 import { useStyles } from 'sku/react-treat';
 import { ContentBlock, Text, Box, Hidden } from '../../../../lib/components';
 import { useIsolatedScroll } from '../../../../lib/components/Autosuggest/useIsolatedScroll';
-import { Overlay } from '../../../../lib/components/private/Overlay/Overlay';
-import { Logo } from '../Logo/Logo';
-import { Link } from '../Link/Link';
-import { MenuButton } from '../MenuButton/MenuButton';
+import { useBoxStyles } from '../../../../lib/components/Box/useBoxStyles';
+import { BoxProps } from '../../../../lib/components/Box/Box';
 import { SubNavigation } from '../SubNavigation/SubNavigation';
+import { MenuButton } from '../MenuButton/MenuButton';
+import { Logo } from '../Logo/Logo';
 import * as styleRefs from './Navigation.treat';
 
+const space = 'medium';
+
 const Header = ({
   menuOpen,
   menuClick,
 }: {
   menuOpen: boolean;
   menuClick: () => void;
 }) => (
-  <Box paddingY={['medium', 'large']}>
-    <Text baseline={false}>
+  <Box paddingY="large" paddingX={space}>
+    <Text component="div" baseline={false}>
       <Box display="flex" alignItems="center">
         <Hidden print above="mobile">
           <Box paddingRight="medium" display="flex" alignItems="center">
             <MenuButton open={menuOpen} onClick={menuClick} />
           </Box>
         </Hidden>
-        <Link to="/" tabIndex={menuOpen ? -1 : undefined}>
+        <Link
+          to="/"
+          tabIndex={menuOpen ? -1 : undefined}
+          className={useBoxStyles({ component: 'a' })}
+        >
           <Logo iconOnly height={32} />
         </Link>
       </Box>
     </Text>
   </Box>
 );
 
+const FixedContentBlock = forwardRef<HTMLElement, BoxProps>(
+  ({ children, ...props }, forwardedRef) => (
+    <Box transition="fast" {...props} position="fixed" ref={forwardedRef}>
+      <ContentBlock width="large">{children}</ContentBlock>
+    </Box>
+  ),
+);
+
 interface NavigationProps {
   children: ReactNode;
 }
+
 export const Navigation = ({ children }: NavigationProps) => {
   const styles = useStyles(styleRefs);
+  const { y: scrollTop } = useWindowScroll();
+  const [showStickyHeader, setShowStickyHeader] = useState(false);
   const [isMenuOpen, setMenuOpen] = useState(false);
 
+  const [lastScrollTop, setLastScrollTop] = useState(0);
+  const [direction, setDirection] = useState<'up' | 'down' | null>(null);
+
+  useInterval(() => {
+    if (lastScrollTop !== scrollTop) {
+      if (Math.abs(lastScrollTop - scrollTop) > 0) {
+        setDirection(scrollTop > lastScrollTop ? 'down' : 'up');
+      }
+      setLastScrollTop(scrollTop);
+    }
+  }, 250);
+
+  const location = useLocation();
+  useEffect(() => {
+    setDirection(null);
+  }, [location]);
+
+  useEffect(() => {
+    setShowStickyHeader(scrollTop > 300 && direction === 'up');
+  }, [direction, scrollTop]);
+
+  useEffect(() => {
+    document.body.classList[isMenuOpen ? 'add' : 'remove'](styles.scrollLock);
```

Comment:
Could maybe refactor this to a hook e.g. `useScrollLock({ locked: isMenuOpen })`? 

Feel like it becomes more self documenting for future reference.

Reply:
Love it!